dace/frontend/python/replacements.py

```diff
@@ -452,58 +453,98 @@ def _argminmax(sdfg: SDFG,
+            for i, n in enumerate(a_arr.shape)
+        },
+        inputs={
+            '__in':
+            Memlet.simple(
+                a, ','.join('__i%d' % i for i in range(len(a_arr.shape))))
+        },
+        code="__out = _val_and_idx(idx={}, val=__in)".format("__i%d" % axis),
+        outputs={
             '__out':
             Memlet.simple(
-                structs,
-                ','.join('__i%d' % i for i in range(len(a_arr.shape))))
+                reduced_structs,
+                ','.join('__i%d' % i for i in range(len(a_arr.shape))
+                         if i != axis),
+                wcr_str=("lambda x, y:"
+                         "_val_and_idx(val={}(x.val, y.val), "
+                         "idx=(y.idx if x.val {} y.val else x.idx))").format(
+                             func, '<' if func == 'max' else '>'))
         },
         external_edges=True)
 
-    # reduce array of structs
-    nest(_reduce)(  # comment for yapf
-        "lambda x, y: _val_and_idx(val={}(x.val, y.val), idx=(x.idx if x.val {} y.val else y.idx))"
-        .format(func, '>' if func == 'max' else '<'),
-        structs,
-        out_array=reduced_structs,
-        axis=axis)
+    if return_both:
+        outidx, outidxarr = sdfg.add_temp_transient(
+            sdfg.arrays[reduced_structs].shape, result_type)
+        outval, outvalarr = sdfg.add_temp_transient(
+            sdfg.arrays[reduced_structs].shape, a_arr.dtype)
+
+        nest.add_state().add_mapped_tasklet(
+            name="_arg{}_extract_".format(func),
+            map_ranges={
+                '__i%d' % i: '0:%s' % n
+                for i, n in enumerate(a_arr.shape) if i != axis
+            },
+            inputs={
+                '__in':
+                Memlet.simple(
+                    reduced_structs, ','.join('__i%d' % i for i in range(len(a_arr.shape)) if i != axis))
+            },
+            code="__out_val = __in.val\n__out_idx = __in.idx",
+            outputs={
+                '__out_val':
+                Memlet.simple(
+                    outval,
+                    ','.join('__i%d' % i for i in range(len(a_arr.shape))
+                             if i != axis)),
+                '__out_idx':
+                Memlet.simple(
+                    outidx,
+                    ','.join('__i%d' % i for i in range(len(a_arr.shape))
+                             if i != axis))
+            },
+            external_edges=True)
 
-    # map to int64
-    out, outarr = sdfg.add_temp_transient(sdfg.arrays[reduced_structs].shape,
-                                          dace.int64)
-    nest(_elementwise)("lambda x: x.idx", reduced_structs, out_array=out)
+        return nest, (outval, outidx)
 
-    return nest, out
+    else:
+        # map to result_type
+        out, outarr = sdfg.add_temp_transient(
+            sdfg.arrays[reduced_structs].shape, result_type)
+        nest(_elementwise)("lambda x: x.idx", reduced_structs, out_array=out)
```

Comment:
@tbennun It would be cool to have memlets that read a field from a struct so that we don't have to copy everything